### PR TITLE
NFDIV-695 Add joint jurisdiction connection

### PR DESCRIPTION
**Pull request by lara-mpt**
### JIRA link ###
https://tools.hmcts.net/jira/browse/NFDIV-695

### Change description ###
Added a jurisdiction connection for the joint journey

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/app/case/definition.ts [hunk 1]
```diff
@@ -374,6 +375,10 @@ export const enum JurisdictionConnections {
    * Applicant 2 is domiciled in England and Wales
    */
   APP_2_DOMICILED = 'I',
+  /**
+   * Applicant 1 habitually resides in England and Wales and joint application
+   */
+  APP_1_RESIDENT = 'J',
```
Comment (velimd) [inline src/main/app/case/definition.ts:381]:
```suggestion
  APP_1_RESIDENT_JOINT = 'J',
```

File: src/main/app/case/definition.ts [hunk 2]
```diff
@@ -469,7 +474,7 @@ export const enum PaymentStatus {
   CANCELLED = 'cancelled',
   ERROR = 'error',
 }
-export const CASE_TYPE = 'NO_FAULT_DIVORCE12';
+export const CASE_TYPE = 'NO_FAULT_DIVORCE14';
```
Comment (velimd) [inline src/main/app/case/definition.ts:477]: this might break stuff as the LegalProcessed property has changed

Comment (velimd) [reply src/main/app/case/definition.ts:477]: so the test might never pass as 2 of the properties were changed, 1 was a minor changed but the other requires abit of thought.

File: src/main/app/case/definition.ts (hunk 1 above)
Comment (lara-mpt) [reply src/main/app/case/definition.ts:381]: Yeah that's a better label, will need to update on backend first though

File: src/main/app/case/definition.ts (hunk 2 above)
Comment (lara-mpt) [reply src/main/app/case/definition.ts:477]: Ah I needed to bump it to get the new jurisdiction field change

Comment (velimd) [reply src/main/app/case/definition.ts:477]: yhh, if you can convert it go for it. Maybe i missed something when looking at it. Tom has a branch you can work off or copy.

Comment (lara-mpt) [reply src/main/app/case/definition.ts:477]: cool, I'll look into it. Worst case could we revert his changes for the time being?

Comment (velimd) [reply src/main/app/case/definition.ts:477]: ideally, i would want that but then im not sure about the repercussion on the solicitor journey, we can also fake some of the data or pass null until its fixed but probably best to look at the ticket or discuss with Tom (hes away until tuesday)

**Review by velimd — APPROVED** [review]

**Pull request merged**